flow/project.py

```diff
@@ -881,25 +1051,30 @@ def _generate_id(self, job, operation_name=None, index=0):
         if max_len < len(job_op_id):
```

Comment:
I am having a bit of difficulty determining if this code will work correctly with large aggregates. Please make sure to check that this code behaves as you expect for large aggregates. The part I'm confused about is the section involving `MAX_LEN_ID` and how it interacts with the parts above and below (whether we have the appropriate guarantees on uniqueness of scheduler ids).

Reply:
I thought `job_op_id` is of constant length for any number of jobs in aggregate.
We calculate the job_op_id in this way
```
    full_name = '{}%{}%{}%{}'.format(project.root_directory(),
                                     '-'.join(map(str, jobs)),
                                     op_string,
                                     index)
    job_op_id = calc_id(full_name)
```
But still, I'll check and let you know the results.

Reply:
Moreover, I think this should be addressed in #306 as we're not yet dealing with Aggregates in this PR.

Reply:
It works fine for an aggregate of length 100k.